### PR TITLE
[WIP] Add cache levels and other settings

### PR DESCRIPTION
**Pull request by hectorj2f**
This PR aims to improve the cache settings of our base nginx image. 

- [ ] test it using our UI apps
- [ ] Questions: I need to ask why some configurations were choosen like `if_modified_since: off` or etag: off

related to: https://github.com/fabric8-ui/fabric8-ui-openshift-nginx/pull/4

### CODE REVIEW

**Comment by fabric8cd** [comment]
@hectorj2f snapshot fabric8-ui-openshift-nginx image is available for testing.  `docker pull fabric8/fabric8-openshift-nginx:SNAPSHOT-PR-4-1`

**Comment by fabric8cd** [comment]
@hectorj2f snapshot fabric8-ui-openshift-nginx image is available for testing.  `docker pull fabric8/fabric8-openshift-nginx:SNAPSHOT-PR-4-1`

File: root/etc/nginx/nginx.conf [hunk 1]
```diff
@@ -48,6 +49,24 @@ http {
     # for more information.
     include /etc/nginx/conf.d/*.conf;
 
+    proxy_cache_path  /etc/nginx/openshiftio_cache levels=1:2 keys_zone=openshiftio_cache:10m max_size=10g inactive=60m use_temp_path=off;
```
Comment (kbsingh) [inline root/etc/nginx/nginx.conf:52]: note the path setup in the dockerfile isnt openshiftio_cache but just cache

Comment (kbsingh) [reply root/etc/nginx/nginx.conf:52]: https://github.com/fabric8-ui/fabric8-ui-openshift-nginx/pull/4/files#diff-3254677a7917c6c01f55212f86c57fbfR18 as ref

**Comment by kbsingh** [comment]
https://github.com/kbsingh/openshift-nginx/tree/proxy has a working cache setup for nginx

**Comment by nurali-techie** [comment]
Thanks @kbsingh for sharing nginx cache example.  I tried running docker image with add_header $upstream_cache_status.  Value of upstream_cache_status was always "MISS".  After some investigation, I found that we need to add "proxy_cache_valid" to nginx.conf.  I have added "proxy_cache_valid 200 10m;" at http level and after that for first request I get value MISS for upstream_cache_status and for second request I get value HIT.  After making this change, I can now see that directory are created under proxy_cache_path with cache content.

**Comment by nurali-techie** [comment]
[test]

**Comment by fabric8cd** [comment]
@hectorj2f snapshot fabric8-ui-openshift-nginx image is available for testing.  `docker pull fabric8/fabric8-openshift-nginx:SNAPSHOT-PR-4-2`

**Comment by nurali-techie** [comment]
The work done here in PR is related to proxy cache.  What we are looking for is browser cache.  So, this PR is not going to merge.  @kbsingh thanks for pointing out on proxy cache.

**Comment by aslakknutsen** [comment]
replaced by https://github.com/fabric8-ui/fabric8-ui-openshift-nginx/pull/17

**Pull request closed without merging**